clang/test/SemaCXX/builtin-std-move.cpp

```diff
@@ -163,6 +170,40 @@ void attribute_const() {
   std::as_const(n); // expected-warning {{ignoring return value}}
 }
 
+struct D {
+  void* operator new(__SIZE_TYPE__, D&&(*)(D&));
+  void* operator new(__SIZE_TYPE__, D*(*)(D&));
+  void* operator new(__SIZE_TYPE__, const D&(*)(D&));
+};
+
+#if __cplusplus <= 201703L
```

Comment:
I realize elsewhere does this... but please use `-verify=cpp20,expected` and `-verify=cpp17,expected`, then make these `cpp17-warning` and `cpp20-error`.

An NFC followup to clean up the other uses here would also be appreciated, but not necessary.  @Endilll has been doing similar cleanups elsewhere.

Reply:
Here's a good example of every kind of `verify` prefixes I've been using in C++ DR tests:
https://github.com/llvm/llvm-project/blob/569ec185f5dc4a9e4a239948191977ecc2b2b475/clang/test/CXX/drs/dr6xx.cpp#L1-L6

Reply:
Alright, done